tests/hypothesis/bin_test.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
So were you only able to cover these edge cases with hypothesis, or can you also add more deterministic tests?

Reply:
I think it could be made more deterministic, yes.

Reply:
See update.